src/main/java/de/tum/in/www1/artemis/domain/exam/StudentExam.java

```diff
@@ -231,6 +246,10 @@ public Boolean isEnded() {
         return ZonedDateTime.now().isAfter(getIndividualEndDate());
     }
 
+    public boolean isFinished() {
+        return Boolean.TRUE.equals(this.isStarted()) && (Boolean.TRUE.equals(this.isEnded()) || Boolean.TRUE.equals(this.isSubmitted()));
```

Comment:
If the is...... methods return booleans anyway, I guess the equals can be omitted

Reply:
not really, these methods return `Boolean`, not `boolean`, which means that they can possibly return `null`